src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -119,6 +119,7 @@ public MgmtOutputLibrary()
             CommonSingleWordModels.Update(_allSchemas);
             NormalizeParamNames.Update(ResourceDataSchemaNameToOperationSets);
             RenameTimeToOn.UpdateNames(_allSchemas);
+            _allSchemas.UpdateDescriptionWithDiscriminator();
```

Comment:
Could we put this method into its own static class with a descriptive name? The extension method pattern does not make much sense because we are not semantically extending method to this object, instead we are changing them. Michael kind of does not like this pattern, therefore you might notice that the decorators added recently are just static methods.

Reply:
Thanks for your review! Actually I'm also adding the description to the property with the base class type. And I have made a change locally to update the decription in the `MgmtObjectType` because:

- I found that we also update the model name elsewhere, thus causing the description update here to not match the final name
- We will apply SecurityElement.Escape() when build the description from schema's description, which leads to the [character replace](https://docs.microsoft.com/en-us/dotnet/api/system.security.securityelement.escape?view=net-6.0#remarks). I don't like this behavior because it will let us miss the compilation failure caused by the wrong name in the summary